ykcompile/src/lib.rs

```diff
@@ -141,17 +151,67 @@ impl Location {
         }
     }
 
-    /// If `self` is a `Register` then unwrap it, otherwise panic.
-    fn unwrap_reg(&self) -> u8 {
-        if let Location::Register(reg) = self {
-            *reg
+    /// If `self` is a `Mem` then return a mutable reference to its innards, otherwise panic.
+    fn unwrap_mem_mut(&mut self) -> &mut RegAndOffset {
+        if let Location::Mem(ro) = self {
+            ro
         } else {
-            panic!("tried to unwrap a Register location when it wasn't a Register");
+            panic!("tried to unwrap a Mem location when it wasn't a Mem");
+        }
+    }
+
+    /// Returns which register (if any) is used in addressing this location.
+    fn uses_reg(&self) -> Option<u8> {
+        match self {
+            Location::Register(reg) => Some(*reg),
+            Location::Mem(RegAndOffset { reg, .. }) => Some(*reg),
+            Location::Indirect {
+                ptr: IndirectLoc::Register(reg),
+                ..
+            } => Some(*reg),
+            Location::Indirect {
+                ptr: IndirectLoc::Mem(RegAndOffset { reg, .. }),
+                ..
+            } => Some(*reg),
+            Location::Const { .. } => None,
+            Location::NotLive => unreachable!(),
+        }
+    }
+
+    /// Apply an offset to the location, returning a new one.
+    fn offset(self, offs: i32) -> Self {
```

Comment:
`offs` -> `off`? [`offs` is an uncommon spelling, and I briefly wondered what it stood for!]

Reply:
We use `offs` all over the place already, and I prefer it as it can't be confused with on/off. Unless you feel strongly, I'd like to keep it.

Reply:
What's the convention in other parts of the codebase e.g. rustc? My strong suspicion is that `off` is more common (I thought `offs` might refer to a set / sequence of `off`s!). Either way, we can do this as a mechanical PR subsequent to this.

Reply:
In rustc they tend to spell it out in full. Same in cranelift.

But at the same time I'm trying to save my hands :P

Reply:
OK, I'm going to leave this one to you to think about. It's not worth us spending too much time on.

Reply:
For what it's worth, I've commented on this as well some time ago as I found `offs` a bit odd, but it's not a hill I'm going to die on.

Reply:
Looks like 2 vs 1. @ptersilie if you are willing to do the change, I will back down.

Reply:
But I'd wait until I've done the OffT change.

Reply:
Pushed. Go ahead with that if you like.

Reply:
Fixed in 1ea1bec . Note that this also requires changes in ykrustc: 5b96b1b